sdks/csharp/tests~/Tests.cs

```diff
@@ -81,4 +81,56 @@ public static void GenericEqualityComparerCheck()
         Assert.False(GenericEqualityComparer.Instance.Equals(statusFailed, statusFailedUnequalValue));
         Assert.False(GenericEqualityComparer.Instance.Equals(statusCommitted, statusOutOfEnergy));
     }
+
+    public class BTreeIndexBaseColumnImplementsIComparableTest
+    {
+
+        public sealed class UserHandle : RemoteTableHandle<EventContext, User>
+        {
+            protected override string RemoteTableName => "user";
+
+            public sealed class IdentityIndex : BTreeIndexBase<SpacetimeDB.Identity>
+            {
+                protected override SpacetimeDB.Identity GetKey(User row) => row.Identity;
+
+                public IdentityIndex(UserHandle table) : base(table) { }
+            }
+
+            public readonly IdentityIndex Identity;
+
+            internal UserHandle(DbConnection conn) : base(conn)
+            {
+                Identity = new(this);
+            }
+        }
```

Comment:
Outstanding comment from the other PR:

> This code won't catch problems that occur if the code generation changes in the future. To catch such problems, we'd need to write the same test against generated code. For example, we could add an index to the Message.Sender field in quickstart-chat -- these tests have access to the quickstart-chat generated code.
> 
> If you want to do that, it would be great; if not, I'm willing to merge this, but would appreciate a comment saying why this class is here. // Copy of generated code for a BTreeIndex; TODO: update if that code changes is messy but at least documents what is going on.